web/webkit/src/main/scala/net/liftweb/http/LiftRules.scala

```diff
@@ -915,6 +915,118 @@ class LiftRules() extends Factory with FormVendor with LazyLoggable {
     ajaxEnd = Full(f: () => JsCmd)
   }
 
+  private [http] val ajaxAddMetaDefaultFunc:JsVar => JsCmd = toSend => Noop
+  private [http] val ajaxQueueSortDefaultFunc:JsVar => JsCmd = ajaxQueue => JE.Call("lift.ajaxQueueSortDefault", ajaxQueue)
+
+  /**
+    * Sets the `ajaxAddMeta()` and `ajaxQueueSort()` javascript functions, allowing the application's ajax
+    * retry logic to be configured.
+    *
+    * By default, Lift will retry failed ajax requests with an exponential backoff (see `LiftRules.ajaxCalcRetryTimeFunc`),
+    * allowing messages to arrive at the server in a shuffled order during periods of failed communications with the server.
+    *
+    * Alternatively, you can have Lift never send messages out of order:
+    * {{{
+    * LiftRules.ajaxAddMetaAndQueueSortFuncs.default.set(() => LiftRules.ajaxAddMetaAndQueueSortInOrderReceivedFuncs)
+    * }}}
+    *
+    * You could also define your own message ordering policy.
+    *
+    * Note: The function you provide is run in the context of a request, allowing you to have conditional logic for
+    * setting these functions.  For instance, you could have on URL which serves an application requiring ordered
+    * messages, while others do not.  Example:
+    * {{{
+    * LiftRules.ajaxAddMetaAndQueueSortFuncs.default.set { () =>
+    *   S.request match {
+    *     case Full(Req("ordered" :: Nil, _, _)) => LiftRules.ajaxAddMetaAndQueueSortInOrderReceivedFuncs
+    *     case _ => LiftRules.ajaxAddMetaAndQueueSortDefaultFuncs
+    *   }
+    * }
+    * }}}
+    *
+    */
+  val ajaxAddMetaAndQueueSortFuncs: FactoryMaker[Box[(JsVar => JsCmd, JsVar => JsCmd)]] = new FactoryMaker[Box[(JsVar => JsCmd, JsVar => JsCmd)]]( () =>
+    Full((ajaxAddMetaDefaultFunc, ajaxQueueSortDefaultFunc))
+  ) {}
+
```

Comment:
I'd love to have some feedback on the design of this setting.  To achieve ordered messages, you have to change both of these functions together.  I think this will be the most common use case.  However, I can imagine where someone may want to change the `ajaxAddMeta` function, but not `ajaxQueueSort`.  For instance, if they wanted to add some meta data related to `ajaxCalcRetryTime` since it has access to the meta.  

I went around in circles a bit and decided to just get it out here and have some feedback.